crates/turbo-tasks-fs/src/lib.rs

```diff
@@ -710,22 +747,29 @@ impl FileSystem for DiskFileSystem {
         let full_path = self.to_sys_path(fs_path).await?;
         let content = content.await?;
 
-        // Track the file, so that we will rewrite it if it ever changes.
-        fs_path.track().await?;
-
         let _lock = self.lock_path(&full_path).await;
 
+        // Track the file, so that we will rewrite it if it ever changes.
+        let old_invalidators = self.register_sole_invalidator(&full_path)?;
+
         // We perform an untracked comparison here, so that this write is not dependent
         // on a read's Vc<FileContent> (and the memory it holds). Our untracked read can
         // be freed immediately. Given this is an output file, it's unlikely any Turbo
         // code will need to read the file from disk into a Vc<FileContent>, so we're
         // not wasting cycles.
         let compare = content.streaming_compare(full_path.clone()).await?;
         if compare == FileComparison::Equal {
+            if !old_invalidators.is_empty() {
```

Comment:
Nit: comment why we're preserving the old invalidators. Especially because it seems like we're trying to emit the same contents from multiple tasks, shouldn't that be an error, too?

Reply:
old_invalidators can also contain invalidators from `read` methods. It's not `write`-only.

We don't want to invalidate read calls when writing identical content to a file.

Reply:
> We don't want to invalidate read calls when writing identical content to a file.

But that means there are two owners of the file. Shouldn't that be an error, why aren't they reusing the same write cell?

Reply:
maybe, but `old_invalidators.is_empty()` is not related to that case, since the invalidator map stores read and write invalidators. Having `old_invalidators` is valid, since they might be invalidators from read methods.